provider/datagen/src/transform/segmenter/mod.rs

```diff
@@ -337,11 +340,23 @@ impl SegmenterRuleProvider {
                             for c in 0..0x20000 {
                                 if lb.get_u32(c) == LineBreak::ComplexContext {
                                     properties_map[c as usize] = property_index
+                                } else if let Some(c) = char::from_u32(c) {
+                                    match script.get(c) {
+                                        Script::Han | Script::Katakana | Script::Hiragana => {
+                                            properties_map[c as usize] = property_index;
```

Comment:
Nit: This change looks like it can be decoupled from the rest of the PR. Please move it to its own PR.

Reply:
OK, I will move mixed text (letter and Chinese) to other PR.